### ISSUE TITLE
accept api-key headers in the livenessProbe, readinessProbe and startupProbe

### ISSUE THREAD

**Issue by bvasil** [qdrant/qdrant-helm#42]
The definition of the livenessProbe, readinessProbe and startupProbe in the statefulset 
don't allow the use of the api-key.

When the qdrant service in initialised with api-key , e.g. 
```
config:
  service:
    api_key: your-qdrant-api-key
```
then the health checks from the probes are failing. 
```
kubectl describe pod qdrant-release-0
```
shows
```
Events:
  Type     Reason                  Age                    From                     Message
  ----     ------                  ----                   ----                     -------
  Normal   Pulled                  6m48s                  kubelet                  Container image "qdrant/qdrant:v1.2.0" already present on machine
  Normal   Created                 6m48s                  kubelet                  Created container qdrant
  Normal   Started                 6m48s                  kubelet                  Started container qdrant
  Warning  Unhealthy               103s (x53 over 6m38s)  kubelet                  Startup probe failed: HTTP probe failed with statuscode: 403
```

An extension of the `statefulset.yaml` to include the `httpHeaders` in the `httpGet` solves this problem once the statefulset is edited manually, e.g.
```
readinessProbe:
  failureThreshold: 6
  httpGet:
    path: /
    port: 6333
    scheme: HTTP
    httpHeaders:
      - name: api-key
        value: your-qdrant-api-key
  initialDelaySeconds: 5
  periodSeconds: 5
  successThreshold: 1
  timeoutSeconds: 1
```

**Comment by bvasil** [comment]
One easy way to fix this could be to add 
```httpHeaders:
        {{- if and $values.config.service.api_key .checksEnabled}}
        - name: api-key
          value: {{ $values.config.service.api_key }}
        {{- end }}
```

at the `livenessProbe`, `readinessProbe` and `startupProbe` in the `statefulset.yaml`

**Comment by timvisee** [comment]
Thank you for reporting this.

I've whitelisted the `/` endpoint in [this](https://github.com/qdrant/qdrant/pull/2407) PR instead, which means it doesn't require an API key anymore. It is what is currently used for these checks. This should fix the health check issue you're describing above, without requiring extra configuration changes.

It is now released as part of [v1.4.1](https://github.com/qdrant/qdrant/releases/tag/v1.4.1). Could you update to this latest version and confirm this is now working?

Note that we're also [adding](https://github.com/qdrant/qdrant/pull/2409) `healthz`, `livez` and `readyz` endpoints, but this is postponed until the next feature release.

**Comment by Zimmerling** [comment]
I had the same problem and with v1.4.1 it is working fine:)
